panel/widgets/select.py

```diff
@@ -654,6 +673,73 @@ def _update_options_programmatically(self):
                 self.value = original_values
                 raise
 
+    @classmethod
+    def create_options_from_dataframe(cls, data: pd.DataFrame, columns: List|None=None)->Dict|List:
+        """Returns the options to use with a NestedSelect
+
+        Args:
+            data (pd.DataFrame): A DataFrame to create options from.
+            columns (List | None, optional): The columns to create options from. If None then all
+                columns are used. Defaults to None.
+
+        Returns:
+            Dict|List: A dictionary (or list) of options. The options are sorted ascending.
+        """
+        if columns is None:
+            columns = list(data.columns)
+
+        data = data[columns].drop_duplicates().sort_values(columns).reset_index(drop=True)
+        options = _build_nested_dict(data)
+        return options
+
+    @classmethod
+    def create_from_dataframe(cls, data: pd.DataFrame, columns: List|None=None, **params)->'NestedSelect':
```

Comment:
Again, if we're going to add something like this we should generalize it and add it to other widgets. Even without having a greater plan though, creating the options from columns is just one approach, creating it from an index is just as valid I would think.